o1vm/src/pickles/column_env.rs

```diff
@@ -0,0 +1,102 @@
+use ark_ff::FftField;
+use ark_poly::{Evaluations, Radix2EvaluationDomain};
+
+use crate::interpreters::mips::witness::SCRATCH_SIZE;
+use kimchi::circuits::{
+    domains::EvaluationDomains,
+    expr::{
+        BerkeleyChallengeTerm, BerkeleyChallenges, ColumnEnvironment as TColumnEnvironment,
+        Constants, Domain,
+    },
+};
+
+use super::proof::WitnessColumns;
+
+/// The collection of polynomials (all in evaluation form) and constants
+/// required to evaluate an expression as a polynomial.
+///
+/// All are evaluations.
+pub struct ColumnEnvironment<'a, F: FftField> {
+    /// The witness column polynomials. Includes relation columns and dynamic
+    /// selector columns.
+    pub witness: &'a WitnessColumns<Evaluations<F, Radix2EvaluationDomain<F>>>,
+    /// The value `prod_{j != 1} (1 - ω^j)`, used for efficiently
+    /// computing the evaluations of the unnormalized Lagrange basis
+    /// polynomials.
+    pub l0_1: F,
+    /// Constant values required
+    pub constants: Constants<F>,
+    /// Challenges from the IOP.
+    // FIXME: change for other challenges
+    pub challenges: BerkeleyChallenges<F>,
+    /// The domains used in the PLONK argument.
+    pub domain: EvaluationDomains<F>,
+}
+
+impl<'a, F: FftField> TColumnEnvironment<'a, F, BerkeleyChallengeTerm, BerkeleyChallenges<F>>
+    for ColumnEnvironment<'a, F>
+{
+    // FIXME: do we change to the MIPS column type?
+    // We do not want to keep kimchi_msm/generic prover
+    type Column = kimchi_msm::columns::Column;
+
+    fn get_column(
```

Comment:
a general comment on all the get functions
I thought we would use the column env as a black box containing 'everything we need to evaluate', and pass that to an evaluation function
Probably I'm missing some details, but I thought we would not need access to all the fields

Reply:
Yes, this environment is given later when computing the quotient polynomial.

Reply:
I meant do we need all theses getters ?
I don't see them used in this PR

Reply:
> I meant do we need all theses getters ?

See the code in expr.